### ISSUE TITLE
Design changes for package and author search results

### ISSUE THREAD

**Issue by daveverwer** [SwiftPackageIndex/SwiftPackageIndex-Server#1246]
As I understand it, currently the search results will only ever return one match for an exact match of package author or keyword @finestructure?

I've designed single keyword and author results, but the design would work for if/when we also match partial usernames and keywords. For example, the author results could say:

```
Looking for packages by Bob, Bobber, or Bobbins? (See all author results)
```

and keywords could be a list of keywords displayed as tags after the "Packages tagged with", again with a link to a separate results page, if one is needed. I don't think we should try and mix up all results into one search. This page should be the "primary" search page, linking off to others if necessary.

<img width="959" alt="My Sketches - 2021-09-06 17 57 41" src="https://user-images.githubusercontent.com/5180/132248724-4632b315-751c-4580-9f9c-ea6523b923a7.png">

**Comment by finestructure** [comment]
> As I understand it, currently the search results will only ever return one match for an exact match of package author or keyword @finestructure?

Yes, that's correct! Changing it in terms of queries should be rather simple (bookend the search term with `%` at the places indicated with `FIXME: bookend with %`) and increase the limit on the query. The problem is more around what should be the sort order - i.e. how do we determine match ranking.
